### PR TITLE
Full release notes for Version 2.0.0

### PR DESCRIPTION
**Pull request by jyaistMap**
All release notes through Build #2524
* @priyankatuteja - please review the `arcgis.learn` notes
* @rhea-j - please review the `arcgis.raster` and submodule notes

### CODE REVIEW

**Review by AtmaMani — APPROVED** [review]
Looks good to me. Thanks @jyaistMap 
@priyankatuteja , @rhea-j  please provide your reviews as well.

File: guide/02-api-overview/release-notes.ipynb [hunk 1]
```diff
@@ -13,6 +13,18 @@
    "source": [
```
Comment (rhea-j) [inline guide/02-api-overview/release-notes.ipynb:13]: <p>Need to mention fixes in PR   - https://github.com/ArcGIS/geosaurus/pull/7687</p><p> </p><ol><li>Made the calendar fied and operator parameters of filter_by_calendar_range, filter_by_attribute respectively,  case insensitive for raster collections created from datastore rasters.</li><li>RasterCollection engine was not getting set properly when engine is set to "arcpy"</li></ol><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1158/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='35'/>

Comment (rhea-j) [inline guide/02-api-overview/release-notes.ipynb:13]: <p>https://github.com/ArcGIS/geosaurus/pull/7674</p><p>arcgis.raster.ImageryLayer</p><p> </p><ol><li>Fix issues in saving an ImageryLayer item as a portal item with on the fly processing by setting for_viz is set to True.</li><li>Fix to honor the rendering rule and mosaic rule set on a portal item when initialising the imagery layer object.</li></ol><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Esri/arcgis-python-api/pull/1158/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

[the remark this answers is not in the document]
Comment (jyaistMap) [reply guide/02-api-overview/release-notes.ipynb:13]: <p>Addressed these, @rhea-j.</p>

[the remark this answers is not in the document]
Comment (jyaistMap) [reply guide/02-api-overview/release-notes.ipynb:13]: <p>Sorry I missed all these, @rhea-j. Added them!</p>

[the remark this answers is not in the document]
Comment (jyaistMap) [reply guide/02-api-overview/release-notes.ipynb:13]: <p>Thanks! Sorry I missed <a href="ArcGIS/geosaurus/pull/7616" target="_blank">#7616</a></p>

Comment (jyaistMap) [reply guide/02-api-overview/release-notes.ipynb:13]: <p>Thanks. Added those notes.</p>

Comment (jyaistMap) [reply guide/02-api-overview/release-notes.ipynb:13]: <p>Added them!</p>

**Review by mohi9282 — APPROVED** [review]
Looks good

**Pull request merged**